packages/studiocms_devapps/src/utils/wp-api/converters.ts

```diff
@@ -76,17 +78,114 @@ export const ConvertToPageContent = async (
 	return pageContent;
 };
 
-export const ConvertToPostData = async (post: unknown, useBlogPkg: boolean): Promise<PageData> => {
+export const generateCatagories = async (categories: number[], endpoint: string) => {
+	const newCatagories: Category[] = [];
+
+	for (const catagoryId of categories) {
+		// Check if catagory already exists in the database
+		const catagoryExists = await db
+			.select()
+			.from(tsPageDataCategories)
+			.where(eq(tsPageDataCategories.id, catagoryId))
+			.get();
+
+		if (!catagoryExists) {
+			const catagoryURL = apiEndpoint(endpoint, 'catagories', `${catagoryId}`);
+			const response = await fetch(catagoryURL);
+			const json = await response.json();
+			newCatagories.push(json);
+		}
+
+		if (catagoryExists) {
+			console.log(`Catagory with id ${catagoryId} already exists in the database`);
```

Comment:
Also is there a reason `console.log` is being used and not the logger?

Reply:
runtime Logger is not configured yet on the devapps(would add another dep to that package), so i've just been using console.log